ITISFoundation/osparc-simcore
# Pipeline status (front)

## What do these changes do?


This PR gives frontend support to new pipeline status functionality.
* Run button responding to pipeline status
* Performs the 60 seconds check if no pipeline status update has been received

#### What I have done:
* Button going to fetching state, it also gets disabled.
* Pressing the button -> Fetching
* Pipeline is saved
  * If saving fails, button Not Fetching
* If pipeline saves, pipeline is started
  * If starting fails, button Not fetching
* Button listens to ``projectStateUpdated`` WS event and updates depending on the reported state.
* A 60 second timer is set to call ``/state`` endpoint and update the button state when no ``projectStateUpdated`` has been received after the ``/start`` call.

## Related issue number


#1781 

## How to test


* Run the pipeline and check run button
* Try to simulate different server events and see that the run button updates accordingly.

## Checklist
